### ISSUE TITLE
QueryAsync<T> Does Not Ever Return If Zero Results

### ISSUE THREAD

**Issue by thyams** [StackExchange/Dapper#769]
It works great when rows are returned but if no data was found...  ...scope never comes back from await.

Did a SQL Profile to a localdb, verifying storedproc hits, validate the call works, returns 0 results instantly manually executed, but Dapper never comes back from QueryAsync. No SQLClient, SQLType, SQL Other Exceptions occur.

Framework 4.6.2
Azure SQL (T-SQL)
Dapper Nuget 1.50.2

Sample Call That Works:
```c#
using( var connection = new SqlConnection( ConnectionString ) )
{
    var result = connection.Query<T>(
        $"[{SchemaString}].[{ObjectType}_{sprocSuffix}]",
        TrimObject( value ),
        commandType: CommandType.StoredProcedure );

    return result.SingleOrDefault();
}
```

Sample Call That Does Not Work:
```c#
using( var connection = new SqlConnection( ConnectionString ) )
{
    var result = await connection.QueryAsync<TId>(
        $"[{SchemaString}].[{ObjectType}_{sprocSuffix}]",
        TrimObject( obj ),
        commandType: CommandType.StoredProcedure );

    return result.SingleOrDefault();
}
```

**Comment by celalit** [comment]
I faced the same issue, but in my case I was calling QueryAsync<> from a syncronised Asp.Net MVC controller. Once, I added 'async' keyword to the Asp.Net MVC controller and 'await' (awaited) the dapper call, it works fine.

**Comment by thyams** [comment]
@celalit Appreciate the response but that isn't what's happening here. In your situation, you were exiting the scope of your MVC controller before the call had a chance to return. In this particular instance, I run into a situation of code frozen forever at an await condition.

**Comment by celalit** [comment]
I see.

**Comment by thyams** [comment]
@NickCraver Any news?

**Comment by eXon** [comment]
This is a huge problem for us as well. The thread is freezing when a `QueryAsync` has no row.

**Comment by 007imash** [comment]
does anyone know when the solution to this issue will be released? I am still experiencing this issue even on the latest pre-release version.

**Comment by 0xfeeddeadbeef** [comment]
I canʼt reproduce this bug on .NET 4.7.1 — it doesnʼt throw (Dapper 1.50.2, SQL Server LocalDB 2016).

**Comment by NickCraver** [comment]
This is fixed on MyGet, but we need to push a NuGet update. I'll try to make it happen this week - we've just been slammed with the launch of Teams and GDPR at Stack Overflow. Hopefully a lot of items off the TODO lists in the coming weeks.

**Comment by thyams** [comment]
Sounds like you need some help, from a fast learning dev ;)

**Comment by NickCraver** [comment]
NuGet is now updated (and I've added MyGet feed data to the Dapper README). Thanks for the help here!

**Comment by HBOlsen** [comment]
@NickCraver I'm having similar problems as described in this issue. What version of Dapper has this fix?

**Comment by vdarshan82** [comment]
On MyGet it shows the version 2.0.51. And its for the .Net Standard Version as per dependencies shown. Does it mean it not implemented for .Net Core 3.0.1. 
Can you confirm whether it will work ?

Also, Can we expect updated version 2.0.51 to be available on nuget soon as it still has v2.0.35 which is old.

**Comment by HasibAhmed577** [comment]
Using Dapper 2.1.35 from .Net Framework 4.8.1, with Fsharp.Core 8.0.400. This issue is still present.
